flang/lib/Lower/OpenMP.cpp

```diff
@@ -882,7 +1009,9 @@ void Fortran::lower::genOpenMPDeclarativeConstruct(
                  "OpenMPDeclareTargetConstruct");
           },
           [&](const Fortran::parser::OpenMPThreadprivate &threadprivate) {
-            TODO(converter.getCurrentLocation(), "OpenMPThreadprivate");
+            // The directive is lowered when instantiating the variable to
+            // support the case of threadprivate variable declared in module.
+            // FIXME: Implement this by adding `thread_local` attribute.
```

Comment:
We favor "hard TODO" messages rather than ignoring features that aren't implemented in general. (Commenting out TODOs when working on something new is arguably easier than producing bad programs or crashing downstream someplace.)

Reply:
This has been implemented and it is what this patch does. Adding FIXME seems confusing and moved it to the place instantiating the variables, where we generate the ThreadprivateOp.